3581/3/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,41 @@
+class UserController < ApplicationController
+  get '/sign_in' do
+    redirect '/' if session?
+    erb :sign_in
+  end
+
+  get '/sign_up' do
+    redirect '/' if session?
+    erb :sign_up
+  end
+
+  get '/logout' do
+    session.clear
+    redirect '/'
+  end
+
+  post '/sign_in' do
+    @user = User.find_by(email: params['email'])
```

Comment:
add the ability to login via name
( you could use `WHERE some_condition OR some_other_condition lIMIT 1`. it is a kind of tip)